### PR TITLE
Update for eval refactor

### PR DESCRIPTION
**Pull request by Innixma**
Issue #, if available:

Description of changes:
- Refactor in prep for overhaul to evaluation pipeline
- Fix pandas warning during eval
- Improve user friendliness and error message verbosity when no valid results after filtering errors
- Simplify raw-to-processed cleaning logic
- Update vs comparison table column names and slightly modify logic

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -40,15 +40,15 @@ dependencies = [
...
     "pydantic>=1.10.4,<2.0", # https://github.com/ray-project/ray/issues/36990
-    "pyyaml >=5.4,<=6.0",
+    "pyyaml >=5.4,<7",
     "ray[default] >=2.6.3,<2.7",
     "s3fs >=2023.5.0,<=2023.6.0",
-    "tqdm >=4.64.0,<=4.65.0",
+    "tqdm >=4.64.0,<=5",
     "typer >=0.9.0,<1.0.0",
-    "wheel >0.38.0,<=0.40.0",
+    "wheel >0.38.0,<0.43",
```
Comment (suzhoum) [inline pyproject.toml:51]: Thanks for updating the dependencies!

**Review by suzhoum — APPROVED** [review]
LGTM! Only left some minor comments.

File: src/autogluon/bench/eval/evaluation/preprocess/preprocess_openml.py [hunk 2]
```diff
@@ -18,19 +26,61 @@ def preprocess_openml_input(
                 framework_rename_dict[key] if framework[0] == key else framework[0]
                 for framework in zip(raw_input[FRAMEWORK])
             ]
-    if framework_suffix_column is not None:
-        if "framework_parent" in raw_input.columns:
-            raw_input["framework_parent"] = raw_input["framework_parent"] + "_" + raw_input[framework_suffix_column]
+
+    def _update_framework_name(name: str, parent: str, name_suffix_1: str, name_suffix_2: str):
```
[the remark this answers is not in the document]
Comment (Innixma) [reply src/autogluon/bench/eval/evaluation/preprocess/preprocess_openml.py]: updated, thanks for the suggestion!

File: src/autogluon/bench/eval/evaluation/preprocess/preprocess_openml.py [hunk 3]
```diff
@@ -18,19 +26,61 @@ def preprocess_openml_input(
...
+
+    raw_input[FRAMEWORK] = updated_names
+    if "framework_parent" in raw_input.columns:
+        raw_input["framework_parent"] = updated_parents
+
+    if "framework_parent" in raw_input:
+        # raw_input["framework_child"] = raw_input[FRAMEWORK]
+        raw_input["framework_child"] = np.where(raw_input["framework_parent"].isnull(), np.nan, raw_input[FRAMEWORK])
+
+    def _parent_rename(parent: str | None, name: str):
```
[the remark this answers is not in the document]
Comment (Innixma) [reply src/autogluon/bench/eval/evaluation/preprocess/preprocess_openml.py]: updated, thanks for the suggestion!

**Pull request merged**